server/api/annotation.py

```diff
@@ -255,6 +274,40 @@ def submitAnnotation(self, annotation, params):
         annotation['stopTime'] = datetime.datetime.utcfromtimestamp(
             bodyJson['stopTime'] / 1000.0)
         annotation['responses'] = bodyJson['responses']
-        annotation['markups'] = bodyJson['markups']
 
-        Annotation().save(annotation)
+        # Before anything is saved, validate all the other (non-'markups') fields
+        Annotation().validate(annotation)
+
+        # Validate superpixel markups
+        study = Study().load(annotation['studyId'], force=True, exc=True)
+        image = Image().load(annotation['imageId'], force=True, exc=True)
+        superpixelsData = Image().superpixelsData(image)
+        maxSuperpixel = int(superpixelsData.max())
+        jsonschema.validate(
```

Comment:
For consistency, since this raises `jsonschema.exceptions.ValidationError`, it would be good to catch that and raise `ValidationException`.